smart-py/marketplace.py

```diff
@@ -53,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
+            sp.verify(self.data.secondary_accounts.contains(sp.sender))
+        sp.else:
+            sp.verify(sp.sender == self.data.manager)
+
         self.data.manager = params
-        
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def add_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.add(params)
```

Comment:
Should there be a check if the new manager is already in the list?

Reply:
I think it's not necessary. We are using a set, so it will not be added twice.